server/src/main/java/com/testsigma/service/TestStepService.java

```diff
@@ -574,8 +577,40 @@ private List<ActionTestDataMap> getMapsList(){
 
         return actionsMap;
     }
+    public void bulkDelete(Long[] ids) throws ResourceNotFoundException {
+        for (Long id:ids){
+              this.destroy(this.find(id));
+            }
+    }
 
-
+    private Boolean checkBulkDeleteStepsHaveAllPrerequisiteSteps(List<Long> ids,List<TestStep> testStepList){
+        for (TestStep testStep:testStepList){
+            if(!ids.contains(testStep.getId())){
+                return false;
+            }
+        }
+        return true;
+    }
```

Comment:
It seems the function just looks for a bool value when there is id exist which is not part of test step list. So, it can be renamed better.
Assume, the ids list will be limited items than no of items in the test step list, if so it's better to iterate over the ids list and check non existence of a step. Instead of iterating over enter step list.